keras_core/layers/layer.py

```diff
@@ -646,7 +643,15 @@ def maybe_convert(x):
                 return backend.convert_to_tensor(x, dtype=self.compute_dtype)
             return x
 
-        if self._convert_input_args:
+        if (
+            not kwargs
+            and len(args) == 1
+            and backend.is_tensor(args[0])
+            and backend.standardize_dtype(args[0].dtype) == self.compute_dtype
+        ):
+            # Used to avoid expensive `tree` operations in the most common case.
+            pass
```

Comment:
Basically no need for `if ...: pass; elif:`, a single `if` is enough

Reply:
Done.